docs/benchmark/ftlib_benchmark.md

```diff
@@ -0,0 +1,171 @@
+
+### ResNet50
+
+There are also 214 trainable tensors in ResNet50 for ImageNet. But the
+model trainable parameters is 23,739,492 because there are 100 classes.
+
+| Workers   | speed | total task time  | allreduce time| tensor.numpy() time| apply_gradients |
+| --------- | ----- | --------------- | -------- | ------ | ---------- |
+| 1 (local) | 168 images/s| 6.1s  |  - | - |  4.16s |
+| 2  | 148 images/s| 13.76s | 10.36s | 5.04s | 1.35s |
+| 4  | 228 images/s| 18s | 14.67s |  5.14s | 1.30s |
+
+![Resnet50](https://user-images.githubusercontent.com/18071380/88752262-0fdd2800-d18c-11ea-8b37-83ba5662e53d.png)
+
+### MobileNetV2
+
+For ImageNet dataset with 100 classes, there are also 158 trainable tensors in
+MobileNetV2 for ImageNet. And those tensors totally have 2,386,084 trainable parameters.
+
+| Workers   | speed | total task time  | allreduce time| tensor.numpy() time| apply_gradients |
+| --------- | ----- | --------------- | -------- | ------ | ---------- |
+| 1 (local) | 169 images/s | 6.06s  |  - | - |  5.59s |
+| 2  | 246 images/s | 8.34s  |   7.25026 | 5.79s | 0.6s |
+| 4  | 401 images/s | 10.2029s | 8.9s |  5.78s | 0.71s |
+
+![MobileNetV2](https://user-images.githubusercontent.com/18071380/88752339-43b84d80-d18c-11ea-9139-c907975a7aeb.png)
+
+### An Image Compression Model with Conv2DTranspose
+
+From the ResNet50 and MobileNetV2 experiments, we can find that
+the speed-up of MobileNetV2 is better than ResNet50. MobileNetV2 has less
+parameter tensors and parameters. In order to research what plays an important
+role in speed-up, we also do exmperiments with an image compression model.
+This model has less parameter tensors and more parameters than MobileNetV2.
+For ImageNet dataset, there are 34 parameter tensors in the model and those
+tensors totally has 11,238,723 parameters.
+
+| Workers   | speed | total task time  | allreduce time| tensor.numpy() time| apply_gradients |
+| --------- | ----- | --------------- | -------- | ------ | ---------- |
+| 1 (local) | 109 images/s | 9.36s  |  - | - |  8.95s |
+| 2  | 176 images/s | 11.65s | 1.47s | 9.36s | 0.42s |
+| 4  | 328 images/s | 12.47s | 2.44s |  9.32s | 0.37s |
+
+![Image Compression](https://user-images.githubusercontent.com/18071380/88752707-3059b200-d18d-11ea-84bd-1db670c64924.png)
+
+For the above experiments of GPU, we find that the speed-up ratio is better if
+the number of parameter tensors is less. Because the computation on GPUs is
+very fast for different models. The communication is the  bottleneck. So the
+model complexity has little effect on the speed-up. The number of parameter
+tensor is less, the communication is more light, so the speed-up is better.
+
+## Conclusion
+
+To improve the performance of FTlib AllReduce in ElasticDL, we need to
+optimize the communication. In order to improve the communication efficiency,
+we can:
+
+1. Interleave communication and computation.
+2. Merge small tensors into a big tensor to reduce communication overhead.
```

Comment:
In case you missed it, these two items among other potential improvements are mentioned in the original design doc as well: https://github.com/sql-machine-learning/elasticdl/blob/develop/docs/designs/allreduce.md#potential-future-optimizations

Reply:
I will add the link.